### PR TITLE
Explicitly set default emails for login/nodeset service account

### PR DESCRIPTION
**Pull request by alyssa-sm**
This PR explicitly sets a default email for login/nodeset service account.

### CODE REVIEW

File: community/modules/compute/schedmd-slurm-gcp-v6-nodeset-tpu/main.tf [hunk 1]
```diff
@@ -50,3 +52,6 @@ locals {
     network_storage = var.network_storage
   }
 }
+
+data "google_compute_default_service_account" "default" {
```
Comment (mr0re1) [inline community/modules/compute/schedmd-slurm-gcp-v6-nodeset-tpu/main.tf:56]: We got to specify `project_id`, please add it as a variable to this module.

File: community/modules/compute/schedmd-slurm-gcp-v6-nodeset/main.tf [hunk 2]
```diff
@@ -96,3 +98,6 @@ locals {
     network_storage   = var.network_storage
   }
 }
+
+data "google_compute_default_service_account" "default" {
```
Comment (mr0re1) [inline community/modules/compute/schedmd-slurm-gcp-v6-nodeset/main.tf:102]: same

**Comment by mr0re1** [comment]
That will do, I'm OK to LGTM it after other comments are addressed.
But, I would strongly consider an alternative solution: fix it in `slurm_instance_template/main.tf`, two reasons:
* there is clearly a bug in `slurm_instance_template`;
* avoid bloating terraform graph with more nodes.

File: community/modules/compute/schedmd-slurm-gcp-v6-nodeset-tpu/main.tf (hunk 1 above)
Comment (alyssa-sm) [reply community/modules/compute/schedmd-slurm-gcp-v6-nodeset-tpu/main.tf:56]: Added.

File: community/modules/compute/schedmd-slurm-gcp-v6-nodeset/main.tf (hunk 2 above)
Comment (alyssa-sm) [reply community/modules/compute/schedmd-slurm-gcp-v6-nodeset/main.tf:102]: Added as well.

**Review by mr0re1 — APPROVED** [review]

**Pull request merged**